src/libstd/net/ip.rs

```diff
@@ -393,8 +393,7 @@ impl Ipv4Addr {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn octets(&self) -> [u8; 4] {
-        let bits = u32::from_be(self.inner.s_addr);
-        [(bits >> 24) as u8, (bits >> 16) as u8, (bits >> 8) as u8, bits as u8]
+        self.inner.s_addr.to_ne_bytes()
```

Comment:
It's surprising to me that there was an explicit big-endian mention here before, but now it's using native-endian.  Obviously the doctest is passing on the little-endian test machine, but are we sure this is correct?  Is it possible that, for clarity, it should be written differently?  Maybe 
```rust
let bits = u32::from_be(self.inner.s_addr);
bits.to_be_bytes()
```
(That should still be efficient, as [LLVM definitely knows that `bswap(bswap(x)) == x`](https://rust.godbolt.org/z/H7xGcS).)

Reply:
Yes, because `s_addr` is always in big endian order and `octets` methods returns data in big endian order too. So we just wanna return bytes as they are and yes, in that case is `to_ne_bytes` little lie. But I am not sure if your proposed solution will more clarify that, maybe write some comment?